libraries/Microsoft.Bot.Builder.Dialogs/Prompts/Prompt.cs

```diff
@@ -22,10 +22,11 @@ public abstract class Prompt<T> : Dialog, IDialogContinue where T : PromptResult
         {
             if (dc == null)
                 throw new ArgumentNullException(nameof(dc));
+
             if (dialogArgs == null)
-                throw new ArgumentNullException(nameof(dialogArgs));
+                throw new Exception("Prompt options are required for Prompt dialogs");
```

Comment:
Please make this code sensible. This should be an ArgumentNullException. If that's not true, then the design is broken and should be fixed as part this PR.

Reply:
... or add a detailed comment around why this behavior is correct.